Objective-C/CBLReplicatorConfiguration.m

```diff
@@ -189,31 +211,95 @@ - (void) setEnableAutoPurge: (BOOL)enableAutoPurge {
     _enableAutoPurge = enableAutoPurge;
 }
 
+- (CBLDatabase*) database {
+    if (!_database)
+        [NSException raise: NSInternalInconsistencyException
+                    format: @"Attempt to access database property, with initWithTarget:"];
```

Comment:
This will only happen when all collections are removed from the collection. You can change the message to something like "Attempt to access database property but no collections added."